integrations/opensearch/src/haystack_integrations/components/retrievers/opensearch/open_search_hybrid_retriever.py

```diff
@@ -0,0 +1,288 @@
+        # DocumentJoiner
+        join_mode: Union[str, JoinMode] = JoinMode.RECIPROCAL_RANK_FUSION,
+        weights: Optional[List[float]] = None,
+        top_k: Optional[int] = None,
+        sort_by_score: bool = True,
+        # extra kwargs
+        **kwargs,
+    ):
+        """
+        Initialize the OpenSearchHybridRetriever, a super component to retrieve documents from OpenSearch using
+        both embedding-based and keyword-based retrieval methods.
+
+        This super component is tied to a SentenceTransformersTextEmbedder.
+
+        We don't explicitly define all the init parameters of the components in the constructor, for each
+        of the components, since that would be around 20+ parameters. Instead, we define the most important ones
+        and pass the rest as kwargs. This is to keep the constructor clean and easy to read.
+
+        If you need to pass extra parameters to the components, you can do so by passing them as kwargs. It expects
+        a dictionary with the component name as the key and the parameters as the value. The component name should be:
+
+            - "bm25_retriever" -> OpenSearchBM25Retriever
+            - "embedding_retriever" -> OpenSearchEmbeddingRetriever
+
+        :param document_store:
+            The OpenSearchDocumentStore to use for retrieval.
+
+        :param embedder:
+            A TextEmbedder to use for embedding the query.
+            See `haystack.components.embedders.types.protocol.TextEmbedder` for more information.
+
+        :param filters_bm25:
+            Filters for the BM25 retriever.
+
+        :param fuzziness:
+            The fuzziness for the BM25 retriever.
+
+        :param top_k_bm25:
+            The number of results to return from the BM25 retriever.
+
+        :param scale_score:
+            Whether to scale the score for the BM25 retriever.
+
+        :param all_terms_must_match:
+            Whether all terms must match for the BM25 retriever.
+
+        :param filter_policy_bm25:
+            The filter policy for the BM25 retriever.
+
+        :param custom_query_bm25:
+            A custom query for the BM25 retriever.
+
+        :param filters_embedding:
+            Filters for the embedding retriever.
+
+        :param top_k_embedding:
+            The number of results to return from the embedding retriever.
+
+        :param filter_policy_embedding:
+            The filter policy for the embedding retriever.
+
+        :param custom_query_embedding:
+            A custom query for the embedding retriever.
+
+        :param join_mode:
+            The mode to use for joining the results from the BM25 and embedding retrievers.
+
+        :param weights:
+            The weights for the joiner.
+
+        :param top_k:
+            The number of results to return from the joiner.
+
+        :param sort_by_score:
+            Whether to sort the results by score.
+
+        :param **kwargs:
+            Additional keyword arguments.
```

Comment:
Could we tell the user what structure we expect for kwargs. It looks like we expect top-level keys to be `bm25_retriever` and `embedding_retriever`

Reply:
Ahh never mind I missed that you provide this explanation up in the description of the docstring. Maybe better to move it here? But up to you.